### PR TITLE
Update for Elm 0.19

### PR DESCRIPTION
**Pull request by xtian**
The elm-community packages aren't updated for 0.19 yet, so I wasn't able to resolve all the compiler errors (and since I couldn't compile, I wasn't able to build the generated code), but I made as much progress as I could. Hope this is useful! :)

### CODE REVIEW

**Comment by dillonkearns** [comment]
@xtian thank you so much for the PR! That's awesome that you did this! I am also going to rename the package to `elm-graphql` with the 0.19 migration, so I will have to make those changes in here as well.

File: package.json [hunk 1]
```diff
@@ -29,16 +29,16 @@
     "elm": "^0.18.0",
     "elm-analyse": "^0.13.3",
     "elm-hot-loader": "0.5.4",
-    "elm-test": "^0.18.12",
     "elm-live": "^2.7.5",
-    "elm-webpack-loader": "^4.3.1",
+    "elm-test": "^0.18.12",
+    "elm-webpack-loader": "git+ssh://git@github.com/xtian/elm-webpack-loader.git#0.19",
```
Comment (xtian) [inline package.json]: https://github.com/elm-community/elm-webpack-loader/pull/142

**Comment by xtian** [comment]
@dillonkearns Got everything compiling, but there are test failures now. I can dig into them in a little bit.

File: src/Graphqelm/Generator/Decoder.elm [hunk 2]
```diff
@@ -56,7 +56,7 @@ generateDecoder apiSubmodule (Type.TypeReference referrableType isNullable) =
             ]
 
         Type.InputObjectRef _ ->
-            Debug.crash "Input objects are only for input not responses, shouldn't need decoder."
+            Debug.todo "Input objects are only for input not responses, shouldn't need decoder."
```
Comment (xtian) [inline src/Graphqelm/Generator/Decoder.elm]: `Debug.todo` is the only equivalent of `Debug.crash`, but the docs state "Note: This is not available with elm make --optimize or packages." so I guess we'll need a different solution in these cases.

**Comment by xtian** [comment]
@dillonkearns Got the tests passing and the generator building. This is ready for review :)

I had to move the generator code into a separate Elm application as the compiler was interpreting the `port module Main` code to be a part of the package and so throwing an error.

**Comment by xtian** [comment]
@dillonkearns You're very welcome! I'm a big fan of this library so I'm happy to help :)

This is back in a non-compiling state unfortunately since the package repo got reset. I'm going to wait and see if new versions get released before republishing my forks.

`elm-community/json-extra` was updated on the alpha repo so it should be released soon. I have [a PR open for `elm-community/string-extra`](https://github.com/elm-community/string-extra/pull/29) but I don't know what the status of that project is.

**Comment by dillonkearns** [comment]
Hey @xtian, I pulled down the code from this PR and was able to get things compiling and the tests passing locally. It looks like it causes a segfault on travis, though, any ideas why?

https://travis-ci.org/dillonkearns/elm-graphql/builds/419360559
https://github.com/dillonkearns/elm-graphql/compare/xtian-elm@0.19?expand=1

Also, I couldn't figure out how to either push commits to this branch directly or make a PR on this PR. Let me know if you have ideas on how to do that, or you could just make these small changes yourself if that's easier.

Also, a couple of things I wanted to discuss:

1. When I run `npm run approve`, I get some changes around the exposing for the modules. Is this something that `elm-format` is doing, or is this in the code generation? https://github.com/dillonkearns/elm-graphql/commit/761eb448e608b7beef2cd54c577f3f0a21510f9f
If it's `elm-format` doing it, then it's totally fine. If were generating it, I'd prefer to keep it simple (if there are no issues that I'm not considering) and just do `module .... exposing (..)` to avoid possible bugs.

2. Can we run the test suite for the `generator` project from the `npm test` task (and by extension in the Travis build)?

I'm making progress on the `dillonkearns/graphqelm` -> `dillonkearns/elm-graphql` rename (I just published [`@dillonkearns/elm-graphql`](https://www.npmjs.com/package/@dillonkearns/elm-graphql), just need to do a little more testing). Then I'll have to do some of those find-and-replaces over to sort out the merge conflicts into this PR, I can take care of all of that. It was tedious but not too bad.

Thanks so much for your help!

**Pull request merged**